### DISCUSSION TITLE
Propose: Add (optional) sub- and supra-national groupings/attributes

### DISCUSSION THREAD

**Discussion by loucasal** [iptv-org/community#1] [Ideas]
Currently the playlist by country is both very long and very uneven in group size.

It is likely that many users would benefit from being able to quickly aggregate sparsely populated groups and to break down the larger ones into more manageable subgroups.

Similarly, users who are more interested in a specific region would probably want a more detailed breakdown there, while a higher level of aggregation would be sufficient for them for other regions.

Obviously this can be customised by each individual user, but it would be more efficient if the main building blocks were provided at the source.

There are probably multiple ways to achieve these goals and I am not too sure whether extra attributes or higher/lower-level groups are the best way to go. I provide some suggestions below as a way to start the discussion - it is not an exhaustive list by any means. 

- To make it easier to aggregate countries, add a region attribute reusing the [region codes](https://github.com/iptv-org/iptv/blob/master/CONTRIBUTING.md#supported-region-codes) already in use within the project, or, alternatively, nest the country groups within their respective continent.
- To make it easier to break down larger country groups (suggestion 1), add a target (or similar) attribute/subgrouping based on whether a channel broadcasts local, national or international content. E.g. think of BBC One Wales vs BBC News vs BBC World News.
- To make it easier to break down larger country groups (suggestion 2), add a network (or similar) attribute/subgrouping based on whether the channel belongs to a specific network/service. E.g. think of the Pluto TV set of channels available in different countries.

What do you think? Would there be any interest in any of the above?

**Comment by BellezaEmporium** [comment]
Sounds interesting for people who wants to get a more detailed / very well structured IPTV playlist.

I would personnally see the playlist like this.

EXAMPLE : index.country.m3u

-> Country -> Region -> Type of channel (General, Entertainment...) -> Channel.

**Comment by freearhey** [comment]
I've read this proposal several times but I still don't understand how exactly it should work? Can someone explain it to me?

**Reply by loucasal** [reply]
I intentionally kept the proposal as broad as possible as a blueprint for a discussion that never happened. And of course it is for you to say whether or not it is feasible at all (nevertheless, I have the impression bits and pieces of this have already been implemented since 2021). However, it can be useful to state a possible use case in a more concrete way.

The underlying assumptions are that (1) users would like to be able to browse channels using a remote, so the number of channels in any given playlist should allow them to do that; and that (2) users' desired level of granularity varies across areas (e.g. high level of granularity for the home country, medium level of granularity for neighbouring countries, low level of granularity for countries that are further away), so flexibility is key.

In an ideal scenario, based on the database and on the attributes described above, users would be able to define the structure of their desired playlist according to their needs.

In a more limited scenario, the repository could provide a few additional breakdowns for users to pick from.

What could the end result look like? An example:

Country 1
- National channels
- Specific network 1 channels
- Specific network n channels
- Local channels

Countries 2 to n
- National channels
- Specific network channels

Regions/Continents 1 to n
- International channels
- National channels

**Reply by freearhey** [reply]
So in short, you're proposing to additionally generate a separate playlist for each network in each country + `local.m3u` + `national.m3u`?

**Reply by loucasal** [reply]
That's one option. Alternatively, to avoid an excessive increase in the number of playlists, national channels could remain in the main country playlists, with only local and network-specific channels moved to a sub-playlist.

**Reply by freearhey** [reply]
Well personally I don't see the benefit of splitting country playlists into smaller files. But maybe others have a different opinion, and someone else will decide to implement this after all.
